### PR TITLE
Format files across the project using pre-commit.

### PR DESCRIPTION
**Pull request by ScottTodd**
Prep for switching our lint checks to use [pre-commit](https://pre-commit.com/) - tracked in https://github.com/iree-org/iree/issues/17430.

Formatting hooks applied in this PR:

* trailing-whitespace
* end-of-file-fixer
* check-yaml
* black
* clang-format
* markdownlint
* bazel_to_cmake

### CODE REVIEW

**Comment by benvanik** [comment]
this seems to be removing end-of-file newlines - I don't think we want that (at least, most editors/git/etc are configured to have them)

File: .devcontainer/devcontainer.json
Comment (ScottTodd) [inline .devcontainer/devcontainer.json:1]: Why is clang-format running on a .json file? o_o
https://github.com/iree-org/iree/actions/runs/9310486480/job/25627953669?pr=17534#step:5:10

File: docs/website/docs/assets/images/iree_architecture.svg
Comment (ScottTodd) [inline docs/website/docs/assets/images/iree_architecture.svg:1]: Argh, thought I excluded images / binary files.

**Comment by ScottTodd** [comment]
> this seems to be removing end-of-file newlines - I don't think we want that (at least, most editors/git/etc are configured to have them)

I think the display in GitHub isn't showing the diffs clearly. The check claims
> ensures that a file is either empty, or ends with one newline.
Source: https://github.com/pre-commit/pre-commit-hooks/blob/main/pre_commit_hooks/end_of_file_fixer.py

Some files have newlines added: 
image

Others have excess newlines removed:
image

For the files with a simple diff I'm wondering what specific end of line characters are used 🤔 (sorta hard to tell on Windows with CRLF git settings):
image

File: tests/e2e/stablehlo_models/generated_e2e_model_tests.cmake
Comment (ScottTodd) [inline tests/e2e/stablehlo_models/generated_e2e_model_tests.cmake:1]: lol the script generating this put a different newline character or something?
https://github.com/iree-org/iree/actions/runs/9310486480/job/25627954248?pr=17534#step:4:10

**Comment by benvanik** [comment]
ah yeah, runtime/readme.md does have an extra newline, but when clicking view file on the change here it's not displayed - seems like a github issue!

File: tests/e2e/stablehlo_models/generated_e2e_model_tests.cmake
Comment (ScottTodd) [reply tests/e2e/stablehlo_models/generated_e2e_model_tests.cmake:1]: May just want to exclude "generated" files (if a regex pattern can detect those without false positives)

File: .devcontainer/devcontainer.json
Comment (ScottTodd) [reply .devcontainer/devcontainer.json:1]: Oh, it supports JSON. Seems fine to format this then https://clang.llvm.org/docs/ClangFormat.html

**Review by kuhar — APPROVED** [review]

**Pull request merged**